spec/requests/graphql/query/publish_target_spec.rb

```diff
@@ -12,7 +12,32 @@
     end
   end
 
-  context 'when logged in user has correct permissions' do
+  context 'when logged in user is admin' do
+    before { sign_in_user as: :administrator }
+
+    context 'when type is valid' do
+      before do
+        graphql query(publish_target.target_type)
+      end
+
+      it 'returns correct response' do
+        expect(response.body).to be_json_eql(%({
+          "project": {
+            "stringKey": "#{project.string_key}",
+            "publishTarget": {
+              "apiKey": "#{publish_target.api_key}",
+              "publishUrl": "https://www.example.com/publish",
+              "type": "PRODUCTION",
+              "doiPriority": 100,
+              "isAllowedDoiTarget": false
+            }
+          }
+        })).at_path('data')
+      end
+    end
+  end
+
+  context 'when logged in user has read permissions' do
```

Comment:
```suggestion
  context 'when logged in user is not an admin, but has read permissions' do
```
(Optional suggestion -- feel free to take it or leave it.)